packages/typescript-estree/tests/lib/createProjectService.test.ts

```diff
@@ -105,16 +111,42 @@ describe('createProjectService', () => {
         defaultProject: './tsconfig.json',
       },
       undefined,
+      undefined,
+    );
+
+    expect(service.setCompilerOptionsForInferredProjects).toHaveBeenCalledWith(
+      compilerOptions,
+    );
+  });
+
+  it('uses tsconfigRootDir as getParsedConfigFile projectDirectory when provided', () => {
+    const compilerOptions = { strict: true };
+    const tsconfigRootDir = 'path/to/repo';
+    mockGetParsedConfigFile.mockReturnValue({ options: compilerOptions });
+
+    const { service } = createProjectService(
+      {
+        allowDefaultProject: ['file.js'],
+        defaultProject: './tsconfig.json',
+      },
+      undefined,
+      tsconfigRootDir,
     );
 
     expect(service.setCompilerOptionsForInferredProjects).toHaveBeenCalledWith(
       compilerOptions,
     );
+    expect(mockGetParsedConfigFile).toHaveBeenCalledWith(
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      require('typescript/lib/tsserverlibrary'),
+      './tsconfig.json',
+      tsconfigRootDir,
+    );
```

Comment:
> what a good test for this would be

Answering in the form of code. It'd be nice if we had an easier integration test setup, but I'm ok with just a unit test for now.